### PR TITLE
protected settings changes

### PR DESCRIPTION
**Pull request by norakoiralamsft**
-- clears out file in config folder (ex: 0.settings)

### CODE REVIEW

File: main/handlersettings_test.go [hunk 1]
```diff
@@ -194,3 +200,48 @@ func Test_toJSONUmarshallForManagedIdentity(t *testing.T) {
...
+		content, err := ioutil.ReadFile(fileName)
+		assert.NoError(t, err)
+		assert.Equal(t, len(content), 0)
+	}
+
+	// cleanup
+	defer os.RemoveAll(testFolderPath)
+}
+
+func createTestFiles(folderPath, settingsExtensionName string) error {
```
Comment (jscalev) [inline main/handlersettings_test.go:227]: should add a test where we have one of the files open so we can't modify it (extension platform should have a similar test whose logic can be copied)

**Review by jscalev — CHANGES_REQUESTED** [review]

File: main/handlersettings_test.go [hunk 2]
```diff
@@ -194,3 +201,62 @@ func Test_toJSONUmarshallForManagedIdentity(t *testing.T) {
...
+	}
+
+	//verify that non settings file did not get cleared
+	fileName = filepath.Join(testFolderPath, "HandlerEnv.txt")
+	content, err := ioutil.ReadFile(fileName)
+	assert.NoError(t, err)
+	assert.Equal(t, len(content), 9)
+
+	// cleanup
+	defer os.RemoveAll(testFolderPath)
```
Comment (D1v38om83r) [inline main/handlersettings_test.go:233]: Can call lines with differ right after the Create TestFiles is called.

File: main/handlersettingscommon.go [hunk 3]
```diff
@@ -140,3 +143,25 @@ func unmarshalProtectedSettings(configFolder string, hs handlerSettingsCommon, v
 	}
 	return nil
 }
+
+// cleanUpSettings replaces the protected settings for all settings files [ex: 0.settings, etc] to ensure no
+// protected settings are logged in VM
+func cleanUpSettings(ctx *log.Context, configFolder string) {
```
Comment (D1v38om83r) [inline main/handlersettingscommon.go:149]: Would Azure-Extension-Platform be a better place to define this function in?

**Comment by norakoiralamsft** [comment]
new PR: https://github.com/Azure/custom-script-extension-linux/pull/201

**Pull request closed without merging**